code/ATMOSPHERICS/hvac/cryotheum_resonator.dm

```diff
@@ -0,0 +1,195 @@
+
+/obj/machinery/cryotheum_resonator
+	density = 1
+	// Thank you Falcon2436 for making the sprites!
+	icon = 'icons/obj/machines/cryotheum_resonator.dmi'
+	icon_state = "machine_active"
+	name = "bluespace cryotheum resonator"
+	desc = "Cutting-edge technology that uses a bluespace crystal to resonate the properties of cryotheum into nearby oxygen."
+	flags = FPRINT
+	machine_flags = EMAGGABLE | WRENCHMOVE | FIXED2WORK | SHUTTLEWRENCH
+	var/activated = FALSE
+	var/obj/item/bluespace_crystal/crystal
+
+/obj/machinery/cryotheum_resonator/examine(mob/user)
+	..()
+	to_chat(user, "<span class='info'>\The [src.name] is currently [activated ? "active" : "inactive"].</span>")
+	if(crystal == null)
+		to_chat(user, "<span class='info'>There is no bluespace crystal installed.</span>")
+	else
+		to_chat(user, "<span class='info'>[crystal] is installed.</span>")
+
+/obj/machinery/cryotheum_resonator/update_icon()
+	overlays.len = 0
+	if(stat & ( NOPOWER | FORCEDISABLE | BROKEN ))
+		//set sprite to  inactive version
+		return
+	else
+		icon_state = "machine_active"
+		if( crystal != null )
+			overlays += image(icon = icon, icon_state = "crystal")
+		if( activated )
+			overlays += image(icon = icon, icon_state = "pulse")
+
+/obj/machinery/cryotheum_resonator/emp_act(severity)
+	if(stat & ( NOPOWER | FORCEDISABLE | BROKEN ))
+		..(severity)
+		return
+
+/obj/machinery/cryotheum_resonator/emag_act(mob/user)
+	if(!emagged)
+		spark(src)
+		to_chat(user, "<span class='warning'>You disable the safety features on \the [src].</span>")
+		log_game("Cryotheum Resonator emagged by [user.ckey]([user]) at ([x],[y],[z]).")
+		emagged = TRUE
+		return 1
+	else
+		spark(src)
+		to_chat(user, "<span class='warning'>You re-enable the safety features on \the [src].</span>")
+		emagged = FALSE
+		return 1
+
+/obj/machinery/cryotheum_resonator/process()
+	if(activated)
```

Comment:
you're indenting a shitton
i would do `if(!activated) return` and continue with the rest of the code on the normal indent level